test/ecto/multi_test.exs

```diff
@@ -336,6 +336,16 @@ defmodule Ecto.MultiTest do
       assert :ok == data.inside_ok
     end
 
+    test "rollbacks on errors in nested function" do
+      multi =
+        Ecto.Multi.new()
+        |> Ecto.Multi.run(:foo, fn repo, _ ->
+          repo.rollback(:bar)
+        end)
+
+      assert {:error, :bar} = TestRepo.transaction(multi)
```

Comment:
should this be the following instead?

```suggestion
      assert {:error, :foo, :bar, %{}} = TestRepo.transaction(multi)
```

this way it's gonna be similar to what happens if any previous multi steps failed.

Moreover, it would return previous steps that did succeed:

```elixir
multi =
  Ecto.Multi.new()
  |> Ecto.Multi.run(:foo, fn repo, _ ->
    {:ok, :good}
  |> Ecto.Multi.run(:bar, fn repo, _ ->
    repo.rollback(:bad)
  end)

{:error, :bar, :bad, %{foo: :good}} = TestRepo.transaction(multi)
```

What do you all think?